packages/firestore/src/local/mutation_queue.ts

```diff
@@ -194,6 +206,17 @@ export interface MutationQueue extends GarbageSource {
     batches: MutationBatch[]
   ): PersistencePromise<void>;
 
+  /**
+   * Clears the cached keys for a mutation batch. This method should be
+   * called by secondary clients after they process mutation updates.
+   *
+   * Note that this method does not have to be called from primary clients as
+   * `acknowledgeBatch()` and `removeMutationBatches()` clear these cache
```

Comment:
I didn't see any code in acknowledgeBatch() that clears the cache...

Reply:
It's done through `removeMutationBatches()`. I reworded this comment as well.